### PR TITLE
Tweak: Prevent dragging a parent container into a child container in Container widget [ED-7559]

### PR DESCRIPTION
**Pull request by hein-obox**

### CODE REVIEW

File: assets/dev/js/editor/elements/views/container.js [hunk 1]
```diff
@@ -147,8 +147,22 @@ const ContainerView = BaseElementView.extend( {
 					newIndex++;
 				}
 
+				// Prevent the user from dragging a parent container into a child container
+				const draggedId = draggedView.getContainer().id;
```
[the remark this answers is not in the document]
Comment (udidol) [reply assets/dev/js/editor/elements/views/container.js:151]: Put the entire block with the `while` loop into an if block that checks this.

File: assets/dev/js/editor/elements/views/container.js [hunk 2]
```diff
@@ -147,8 +147,22 @@ const ContainerView = BaseElementView.extend( {
...
+
+				let draggingParent = false,
```
Comment (udidol) [inline assets/dev/js/editor/elements/views/container.js:153]: move this line out of the new `if` block

**Review by udidol — CHANGES_REQUESTED** [review]
Great job

File: assets/dev/js/editor/elements/views/container.js [hunk 3]
```diff
@@ -147,8 +147,22 @@ const ContainerView = BaseElementView.extend( {
 					newIndex++;
 				}
 
+				// Prevent the user from dragging a parent container into a child container
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply assets/dev/js/editor/elements/views/container.js]: updated

File: assets/dev/js/editor/elements/views/container.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hein-obox) [reply assets/dev/js/editor/elements/views/container.js]: updated

File: assets/dev/js/editor/elements/views/container.js (hunk 2 above)
Comment (hein-obox) [reply assets/dev/js/editor/elements/views/container.js]: I don't think that is necessary, but you can double check that for me.

File: assets/dev/js/editor/elements/views/container.js [hunk 4]
```diff
@@ -149,16 +149,32 @@ const ContainerView = BaseElementView.extend( {
...
-					} );
+					// Prevent the user from dragging a parent container into its own child container
+					const draggedId = draggedView.getContainer().id;
+
+					let draggingParent = false,
+						currentTargetParentContainer = this.container;
+
+					while ( ! draggingParent && currentTargetParentContainer ) {
+						if ( currentTargetParentContainer.id === draggedId ) {
+							draggingParent = true;
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply assets/dev/js/editor/elements/views/container.js]: updated

**Review by udidol — APPROVED** [review]

**Pull request merged**